### PR TITLE
Evaluate collected hardware info

### PR DESCRIPTION
**Pull request by ydayagi**
Do not send unnecessary hadware information
Keep the API as is for now

https://issues.redhat.com/browse/ECOPROJECT-327
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

**Review by jakub-dzon — COMMENTED** [review]
Will it be possible to refactor the code in such a way that we don't even collect unnecessary hardware information? Instead of using `inventory.ReadInventory`, we could use underlying components directly (`inventory.GetCPU`, `inventory.GetHostname`, `inventory.GetVendor`, etc.) - it looks quite easy and could save some CPU cycles.

File: internal/hardware/hardware.go [hunk 1]
```diff
@@ -14,16 +13,17 @@ func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
...
+
+	hardwareInfo.CPU = &models.CPU{
+		Architecture: inv.CPU.Architecture,
+		ModelName:    inv.CPU.ModelName,
+		Flags:        []string{},
 	}
 
+	hardwareInfo.Hostname = inv.Hostname
+
+	hardwareInfo.SystemVendor = (*models.SystemVendor)(inv.SystemVendor)
```
Comment (jakub-dzon) [inline internal/hardware/hardware.go:26]: GPU info and flags details are missing (I understand they were removed in https://issues.redhat.com/browse/ECOPROJECT-327 for clarity and not as a requirement of removal) @masayag - is that correct?

Comment (ydayagi) [reply internal/hardware/hardware.go:26]: I understood that it should be removed. But let's wait for @masayag cause he was in the meeting.

**Comment by ydayagi** [comment]
> Will it be possible to refactor the code in such a way that we don't even collect unnecessary hardware information? Instead of using `inventory.ReadInventory`, we could use underlying components directly (`inventory.GetCPU`, `inventory.GetHostname`, `inventory.GetVendor`, etc.) - it looks quite easy and could save some CPU cycles.

sure. I'll try. if it works then great

File: internal/hardware/hardware.go [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
 }
 
 func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
-	inv := inventory.ReadInventory(&inventory.Options{
-		GhwChrootRoot: "/",
-	})
+	d := util.NewDependencies("/")
+	cpu := inventory.GetCPU(d)
+	hostname := inventory.GetHostname(d)
+	systemVendor := inventory.GetVendor(d)
```
Comment (masayag) [inline internal/hardware/hardware.go:16]: as we learned, we'll have to collect and report IP addresses as well.

Comment (ydayagi) [reply internal/hardware/hardware.go:16]: fixed

File: internal/hardware/hardware.go (hunk 1 above)
Comment (ydayagi) [reply internal/hardware/hardware.go]: @masayag

Comment (masayag) [reply internal/hardware/hardware.go]: yes, they should be removed.

**Review by jakub-dzon — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**